### ISSUE TITLE
Hello im using undetectable chrome driver and it works perfectly except when im using it with with multipale theards it only interacts with 1 theard only which is the first one does anybody knows why ?!

### ISSUE THREAD

**Issue by Virusback** [ultrafunkamsterdam/undetected-chromedriver#682]

**Comment by Virusback** [comment]
@ultrafunkamsterdam

**Comment by Wamy-Dev** [comment]
Please do not @ the dev. Here is the problem from what I know. Threading in python, and in other languages for that matter, is extremely finicky. A lot of modules only work on one thread to begin with, and most modules have to be built with threading in mind, which I suspect chrome driver itself, is not.

**Comment by Virusback** [comment]
So is there is a way to pass this problem or its not possible

**Comment by Virusback** [comment]
> Please do not @ the dev. Here is the problem from what I know. Threading in python, and in other languages for that matter, is extremely finicky. A lot of modules only work on one thread to begin with, and most modules have to be built with threading in mind, which I suspect chrome driver itself, is not. 


So is there is a way to pass this problem or its not possible

**Comment by Wamy-Dev** [comment]
based on my experience and what I know about threading, I don't think it is possible. I don't even think you can do it with normal selenium.

**Comment by DGaffney** [comment]
Yeah, I would run multiple instances and control concurrency outside the scope of the execution of this code. This is almost certainly a recipe for disaster

**Comment by Virusback** [comment]
> based on my experience and what I know about threading, I don't think it is possible. I don't even think you can do it with normal selenium.

no it works peferctly with the normal selenium but the normal selenium dosent provide gmail login

**Comment by sebdelsol** [comment]
It works perfectly with undetected-chromedriver too : I use multi-thread a lot without any issues.

**Please stick to best practices for bugs reporting** : you're supposed to show the **_minimum_ running** code to reproduce the issue and not a wall of barely formatted code with parts missing to make it runs.
